src/rules/import-alias.ts

```diff
@@ -10,23 +10,56 @@ import type { JSONSchema4 } from "json-schema";
 import { dirname, join as joinPath, resolve, sep as pathSep } from "path";
 import slash from "slash";
 
+interface ConfigMatch {
+    type: "pattern" | "path";
+    depth: number;
+}
+
 function isPermittedRelativeImport(
     importModuleName: string,
     relativeImportOverrides: RelativeImportConfig[],
-    fileAbsoluteDir: string
+    filepath: string
 ) {
-    const configOfPath = relativeImportOverrides.find((config) =>
-        fileAbsoluteDir.includes(resolve(config.path))
-    );
-    if (!configOfPath) {
-        return false;
-    }
-
     const importParts = importModuleName.split("/");
     const relativeDepth = importParts.filter(
         (moduleNamePart) => moduleNamePart === ".."
     ).length;
-    return configOfPath.depth >= relativeDepth;
+
+    const relativeMatch: ConfigMatch[] = [];
+
+    relativeImportOverrides.filter((config) => {
+        if ("pattern" in config) {
+            const regex = new RegExp(config.pattern);
+            if (regex.test && regex.test(filepath)) {
+                relativeMatch.push({
+                    type: "pattern",
+                    depth: config.depth,
+                });
+            }
+        }
+
+        if ("path" in config) {
+            if (dirname(filepath).includes(resolve(config.path))) {
+                relativeMatch.push({
+                    type: "path",
+                    depth: config.depth,
+                });
+            }
+        }
+    });
+
+    if (!relativeMatch.length) {
+        return false;
+    }
+
+    const sortedRelativeMatch = relativeMatch.sort((a, b) => a.depth - b.depth);
```

Comment:
I'm okay with this implementation; however, you could also consider simplifying/optimizing it a bit by

1. Sorting the override configurations by depth first. You can do this before passing it into this function when we first construct the array of configs (and also doing it within this function to ensure the invariant in case anything changes in the future since it will be O(n) on sorted arrays anyways).
2. Iterate through the configs
  a. If the path or pattern matches, return true
3. Nothing matched, so return false